src/components/Balance/ProfitLossAnalysis/ProfitLossAnalysis.js

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useState } from "react";
+import "./ProfitLossAnalysis.scss";
+import { Box, Typography, CircularProgress } from "@material-ui/core";
+import { FormattedMessage } from "react-intl";
+import { createEmptyUserEquityEntity } from "../../../services/tradeApiClient.types";
+import { formatNumber } from "../../../utils/formatters";
+
+/**
+ * @typedef {import("../../../services/tradeApiClient.types").DefaultDailyBalanceEntity} DefaultDailyBalanceEntity
+ * @typedef {Object} DefaultProps
+ * @property {DefaultDailyBalanceEntity} dailyBalance Daily balance.
+ */
+
+/**
+ * @param {DefaultProps} props Default props.
+ * @returns {JSX.Element} Component JSX.
+ */
+const ProfitLossAnalysis = ({ dailyBalance }) => {
+  const inittialValues = {
+    totalProfit: 0,
+    totalLoss: 0,
+    netProfitLoss: 0,
+    winDays: 0,
+    lossDays: 0,
+    evenDays: 0,
+    winRate: 0,
+  };
+
+  const [profitLossData, setProfitLossData] = useState(inittialValues);
+
+  const prepareData = () => {
+    let data = { ...inittialValues };
+    const currentEquity = dailyBalance.balances.length
+      ? dailyBalance.balances[dailyBalance.balances.length - 1]
+      : createEmptyUserEquityEntity();
+    [...dailyBalance.balances].forEach((item) => {
```

Comment:
No need destructuring here, `dailyBalance.balances.forEach` is fine.

Reply:
sure yeah.